packages/core/src/framework/history/history-manager.class.ts

```diff
@@ -1,40 +1,52 @@
-import { LoadedEventEmitter, Subscribable } from '../event-emitters';
-import { ElapsedAuguryEvent, ProcessedAuguryEvent } from '../events';
+import { AuguryEvent } from '../events';
 
 export class HistoryManager {
-  public elapsedEvents: ElapsedAuguryEvent[] = [];
+  public elapsedEvents: AuguryEvent[] = [];
 
-  public addEvent(processedEvent: ProcessedAuguryEvent) {
-    const currentPerformanceStamp = performance.now();
-
-    this.elapsedEvents.push({
-      ...processedEvent,
-      completedAtTimestamp: currentPerformanceStamp,
-      auguryDrag: currentPerformanceStamp - processedEvent.creationAtPerformanceStamp,
-    });
+  public addEvent(event: AuguryEvent) {
+    this.elapsedEvents.push(event);
   }
 
   public clear() {
     this.elapsedEvents = [];
   }
 
-  // @todo: ensure startEventId and endEventId are valid
-  public getTotalAuguryDrag(startEventId: number, endEventId: number) {
+  public scan(startEventId: number, endEventId: number) {
```

Comment:
I think at some point it might be good to split this concern out so the history manager can be like a persistence layer.

Reply:
Yep! Future PR next week!